lib/jekyll-admin/static_directory.rb

```diff
@@ -0,0 +1,89 @@
+module JekyllAdmin
+  class StaticDirectory
+
+    include Enumerable
+    include JekyllAdmin::URLable
+    include JekyllAdmin::APIable
+
+    DIRECTORY_TYPE = 'directory'
+    DIRECTORY_SEPARATOR = '/'
+
+    def initialize(path, config={})
+      @base = Pathname.new config.fetch(:base_directory)
+      @path           = Pathname.new path
+      @entries        = initialize_entries
+    end
+
+    def each
+      @entries.each do |entry|
+        yield entry
+      end
+    end
+
+    def to_liquid
+      {
+        basename: basename,
+        name: name,
+        extname: extname,
+        modified_time: modified_time,
+        path: relative_path,
+        type: DIRECTORY_TYPE,
+        entries: entries_to_array
```

Comment:
We'll want to be smart about not recursively responding with all files in a path, but only the current level (to scale on large sites). I'd either use, or base this off the existing `include_content` and `include_documents` arguments in https://github.com/jekyll/jekyll-admin/blob/master/lib/jekyll-admin/apiable.rb#L25.